File: src/main/java/dev/todaka/robustredis/cluster/ClusterNodeView.java
package dev.todaka.robustredis.cluster;

import lombok.Value;

import java.util.List;
import java.util.Set;

/**
 * Represents CLUSTER NODES response line.
 */
@Value
public class ClusterNodeView {
    String id;
    String ip;
    int port;
    Set<String> flags;
    String master;
    String linkState;
    List<Slot> slots;

    @Value
    public static class Slot {
        int from;
        int to;
    }
}

File: src/main/kotlin/dev/todaka/robustredis/cluster/ClusterNodeView.kt
package dev.todaka.robustredis.cluster

data class ClusterNodeView(
    val id: String,
    val ip: String,
    val port: Int,
    val flags: Set<String>,
    val master: String,
    val linkState: String,
    val slots: List<Slot>,
) {
    data class Slot(
        val from: Int,
        val to: Int,
    )
}
